common/lib/dependabot/file_updaters/vendor_updater.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Dependabot
+  module FileUpdaters
+    class VendorUpdater
+      # notable filenames without a reliable extension:
+      TEXT_FILE_NAMES = [
+        "README",
+        "LICENSE",
+        "Gemfile",
+        "Gemfile.lock",
+        ".bundlecache",
+        ".gitignore"
+      ].freeze
+
+      TEXT_FILE_EXTS = [
+        # code
+        ".rb",
+        ".erb",
+        ".gemspec",
+        ".js",
+        ".html",
+        # config
+        ".json",
+        ".xml",
+        ".toml",
+        ".yaml",
+        ".yml",
+        # docs
+        ".md",
+        ".txt",
+        ".go"
+      ].freeze
+
+      def initialize(repo_contents_path:, vendor_dir:)
+        @repo_contents_path = repo_contents_path
+        @vendor_dir = vendor_dir
+      end
+
+      # Returns changed files in the vendor/cache folder
+      #
+      # @param base_directory [String] Update config base directory
+      # @return [Array<Dependabot::DependencyFile>]
+      def updated_vendor_cache_files(base_directory:)
+        return [] unless repo_contents_path && vendor_dir
+
+        Dir.chdir(repo_contents_path) do
+          relative_dir = vendor_dir.sub("#{repo_contents_path}/", "")
```

Comment:
Can we reply on `repo_contents_path` _not_ containing a trailing `/`? I ask because if it does include a trailing `/`, then I think this command would be a no-op, and I'm not sure if that's OK or not.

Reply:
Good question, I think we can, this is a static directory that we configure in the runner: https://github.com/github/dependabot-update-job-runner/blob/13c121542a11ffeb7aaa557178f5b8a4568e4b94/guest/fetcher.go#L20

The diff for that refactor is better viewed here: https://github.com/dependabot/dependabot-core/pull/2598/files note that we've been using that relative dir for bundler vendoring support in prod for a while.

I agree it's a little brittle though and there's room for improvement 👍